### PR TITLE
Refactor status

### PR DESCRIPTION
**Pull request by alxdarksage**
Changing status to reflect the fact there are multiple pathways to authenticate. If the account has at least one valid pathway it's enabled, otherwise it is unverified. Disabled overrides this and sets the account as disabled.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccount.java
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccount.java:379]: Do we have unit tests for this?

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/services/AccountService.java]: Do we have unit tests for this?

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/services/AccountService.java:324]: Do we have a test case for an unverified account but emailVerificationEnabled is false?

File: src/main/java/org/sagebionetworks/bridge/services/EnrollmentService.java
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/services/EnrollmentService.java]: Unit test for caller not in study?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccount.java
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccount.java:379]: Added tests.

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/AccountService.java]: Added

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/AccountService.java:324]: added a case for authenticate and for reauthenticate

File: src/main/java/org/sagebionetworks/bridge/services/EnrollmentService.java
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/EnrollmentService.java]: This is tested by `enrollNotAuthorizedAsStudyResearcher()`

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/AccountService.java]: I also changed the name to be a little clearer that the issue is filtering...if it were just a check-and-throw we could add "self" to the list of permissions, but the filtering is not so easily defeated.

**Comment by alxdarksage** [comment]
This PR has been updated and is ready for a second review.

**Review by DwayneJengSage — APPROVED** [review]

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**
